### PR TITLE
[as4630-54pe] Add to support AP2.0

### PR DESCRIPTION
**Pull request by jostar-yang**
**- Why I did it**
Support API2.0
**- How I did it**
Implement API2 code
**- How to verify it**
root@sonic:/home/admin# ./test_as4630-54pe_api2.py
DEBUG:root:Test Chassis
DEBUG:root:get_name=Accton-AS4630-54PE
DEBUG:root:get_base_mac=1C:EA:0B:89:66:80
DEBUG:root:get_serial_number=463054PE1936004
DEBUG:root:get_system_eeprom_info={'0xFE': '0x28028DEB', '0x28': 'x86_64-accton_as4630_54pe-r0', '0x29': '2018.11.00.03', '0x22': 'F0PZZ4654037A', '0x23': '463054PE1936004', '0x21': 'AS4630-54PE-O-AC-F', '0x27': 'R0C', '0x24': '1C:EA:0B:89:66:80', '0x25': '10/08/2019', '0x2B': 'Accton', '0x2C': 'TW', '0x2A': '256', '0x2D': 'Edgecore', '0x2E': '01.01.01.00'}
DEBUG:root:get_reboot_cause=('REBOOT_CAUSE_NON_HARDWARE', "User issued 'reboot' command [User: admin, Time: Thu 14 Feb 2019 11:29:22 AM UTC]")
DEBUG:root:
Test Component
DEBUG:root:get_num_components=2
DEBUG:root:get_name=CPLD1
DEBUG:root:get_name=BIOS
DEBUG:root:get_firmware_version0 =2
DEBUG:root:get_firmware_version1 =v37.0b.01.01
DEBUG:root:get_description=CPLD 1
DEBUG:root:get_description=Basic Input/Output System
DEBUG:root:
Fan test
DEBUG:root:fan-1, presence=0
DEBUG:root:fan 1 direction=exhaust
DEBUG:root:fan 1 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:fan-2, presence=0
DEBUG:root:fan 2 direction=exhaust
DEBUG:root:fan 2 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:fan-3, presence=0
DEBUG:root:fan 3 direction=intake
DEBUG:root:fan 3 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:fan-4, presence=0
DEBUG:root:fan 4 direction=intake
DEBUG:root:fan 4 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:fan-5, presence=0
DEBUG:root:fan 5 direction=intake
DEBUG:root:fan 5 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:fan-6, presence=0
DEBUG:root:fan 6 direction=intake
DEBUG:root:fan 6 speed()=0
DEBUG:root:set speed to fan-1
DEBUG:root:fan-1 speed()=0
DEBUG:root:
PSU test
DEBUG:root:PSU-1
DEBUG:root:presence=1
DEBUG:root:get_name=PSU-1
DEBUG:root:get_powergood=1
DEBUG:root:get_voltage=54.56
DEBUG:root:get_current=1.11
DEBUG:root:get_power=60.00
DEBUG:root:get_temperature=28.000
DEBUG:root:get_voltage_high_threshold=56.12
DEBUG:root:get_voltage_low_threshold=52.81
DEBUG:root:psu:get_num_fans=1
DEBUG:root:psu:get_fan().get_presence=1
DEBUG:root:psu:get_fan().get_direction=exhaust
DEBUG:root:psu:get_fan().get_speed=47
DEBUG:root:psu:get_temperature=28.000
DEBUG:root:PSU-2
DEBUG:root:presence=1
DEBUG:root:get_name=PSU-2
DEBUG:root:get_powergood=0
DEBUG:root:get_voltage=0.00
DEBUG:root:get_current=0.00
DEBUG:root:get_power=0.00
DEBUG:root:get_temperature=38.000
DEBUG:root:get_voltage_high_threshold=56.12
DEBUG:root:get_voltage_low_threshold=52.81
DEBUG:root:psu:get_num_fans=1
DEBUG:root:psu:get_fan().get_presence=1
DEBUG:root:psu:get_fan().get_direction=exhaust
DEBUG:root:psu:get_fan().get_speed=0
DEBUG:root:psu:get_temperature=38.000
DEBUG:root:
Thermal test
DEBUG:root:Thermal-1
DEBUG:root:get_temperature=44.500
DEBUG:root:high_threshold=85.000
DEBUG:root:get name=Temp sensor 1
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:Thermal-2
DEBUG:root:get_temperature=36.000
DEBUG:root:high_threshold=84.000
DEBUG:root:get name=Temp sensor 2
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:Thermal-3
DEBUG:root:get_temperature=27.000
DEBUG:root:high_threshold=83.000
DEBUG:root:get name=Temp sensor 3
DEBUG:root:get_presence=True
DEBUG:root:get_status=True
DEBUG:root:
SFP test
DEBUG:root:Port-49
DEBUG:root:platform_chassis.get_serial=C1010110933
DEBUG:root:get_model()=HP27008
DEBUG:root:get_name()=Ethernet48
DEBUG:root:get_rx_los=0
DEBUG:root:get_tx_fault()=0
DEBUG:root:get_tx_disable()=0
DEBUG:root:get_lp_mode=0
DEBUG:root:get_tx_disable_channel()=0x0
DEBUG:root:get_temperature=N/A
DEBUG:root:get_voltage=N/A
DEBUG:root:get_tx_bias=['N/A', 'N/A', 'N/A', 'N/A']
DEBUG:root:get_rx_power=['N/A', 'N/A', 'N/A', 'N/A']
DEBUG:root:get_rx_power=['N/A', 'N/A', 'N/A', 'N/A']
DEBUG:root:get_reset_status()=0
DEBUG:root:Port-53
DEBUG:root:platform_chassis.get_serial=APF14120026BUH
DEBUG:root:get_model()=599940002
DEBUG:root:get_name()=Ethernet52
DEBUG:root:get_rx_los=False
DEBUG:root:get_tx_fault()=False
DEBUG:root:get_tx_disable()=[True, True, True, True]
DEBUG:root:get_lp_mode=0
DEBUG:root:get_tx_disable_channel()=0xf
DEBUG:root:get_temperature=0.0
DEBUG:root:get_voltage=0.0
DEBUG:root:get_tx_bias=[0.0, 0.0, 0.0, 0.0]
DEBUG:root:get_rx_power=['N/A', 'N/A', 'N/A', 'N/A']
DEBUG:root:get_rx_power=['N/A', 'N/A', 'N/A', 'N/A']
DEBUG:root:get_reset_status()=0
**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__all__ = ['chassis', 'eeprom', 'platform', 'psu', 'sfp', 'thermal', 'fan']
+from . import platform
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/__init__.py:2]: Add newline at EOF

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/chassis.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        if not self.sfp_module_initialized:
+            self.__initialize_sfp()
+
+        try:
+            # The index will start from 1
+            sfp = self._sfp_list[index-1]
+        except IndexError:
+            sys.stderr.write("SFP index {} out of range (1-{})\n".format(
+                             index, len(self._sfp_list)))
+        return sfp
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/chassis.py:197]: Add newline at EOF

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/component.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        Install firmware to module
+        Args:
+            image_path: A string, path to firmware image
+        Returns:
+            A boolean, True if install successfully, False if not
+        """
+        raise NotImplementedError 
+
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/component.py:125]: Remove trailing blank lines and ensure file ends with a newline char

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/eeprom.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    def get_eeprom(self):
+        return self._eeprom
+
+    def get_serial(self):
+        return self._eeprom.get('0x23', "Undefined.")
+
+    def get_mac(self):
+        return self._eeprom.get('0x24', "Undefined.")
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/eeprom.py:106]: Remove trailing blank lines and ensure file ends with a newline char

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/fan.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        present_path = "{}{}{}".format(CPLD_I2C_PATH, 'present_', self.fan_index+1)
+        val=self._api_helper.read_txt_file(present_path)
+        if not self.is_psu_fan:
+            if val is not None:
+                return int(val, 10)==1
+            else:
+                return False
+        else:
+            return True
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/fan.py:179]: Remove trailing blank lines and ensure file ends with a newline char

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/helper.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            p = subprocess.Popen(
+                cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            raw_data, err = p.communicate()
+            if err == '':
+                result = raw_data.strip()
+            else:
+                status = False
+        except Exception:
+            status = False
+        return status, result
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/helper.py:119]: Add newline at EOF

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/psu.py [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+        Returns:
+            A boolean value, True if device is operating properly, False if not
+        """
+        power_path="{}{}".format(self.cpld_path, 'psu_power_good')
+        val=self._api_helper.read_txt_file(power_path)
+        if val is not None:
+            return int(val, 10) == 1
+        else:
+            return 0
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/psu.py:232]: Remove trailing blank lines and ensure file ends with a newline char

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/thermal.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        """
+
+        file_str = "temp{}_input".format(self.ss_index)
+        file_path = os.path.join(self.hwmon_path, file_str)
+        raw_txt = self.__read_txt_file(file_path)
+        if raw_txt is None:
+            return False
+        else:     
+            return int(raw_txt) != 0
+
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/thermal.py:146]: Remove trailing blank lines and ensure file ends with a newline char

File: platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py [hunk 9]
```diff
@@ -214,10 +219,12 @@ def log_os_system(cmd, show):
     return  status, output
 
 def driver_inserted():
-    ret, lsmod = log_os_system("lsmod| grep accton", 0)
+    ret, lsmod = log_os_system("ls /sys/module/*accton*", 0)
     logging.info('mods:'+lsmod)
-    if len(lsmod) ==0:
+    if ret :
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py:224]: Delete space before `:`

File: platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py [hunk 10]
```diff
@@ -214,10 +219,12 @@ def log_os_system(cmd, show):
...
         return False
+    else :
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py:226]: Delete space before `:`

File: platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py [hunk 11]
```diff
@@ -340,6 +347,73 @@ def system_ready():
         return False
     return True
 
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+PLATFORM_API2_WHL_FILE_PY2 ='sonic_platform-1.0-py2-none-any.whl'
+PLATFORM_API2_WHL_FILE_PY3 ='sonic_platform-1.0-py3-none-any.whl'
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-accton/as4630-54pe/utils/accton_as4630_54pe_util.py:352]: All applications which import sonic_platform should now support Python 3 (as of a couple weeks ago). Therefore, I believe there should no longer be a need to build/install the Python 2 package. Would you mind confirming, and if so, remove all references to the Python 2 package in this PR?

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/chassis.py [hunk 12]
```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/chassis.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/component.py [hunk 13]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/component.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/eeprom.py [hunk 14]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/eeprom.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/fan.py [hunk 15]
```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/fan.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/helper.py [hunk 16]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/helper.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/platform.py [hunk 17]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/platform.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/psu.py [hunk 18]
```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/psu.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py [hunk 19]
```diff
@@ -0,0 +1,1169 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py:1]: Script is not executable; shebang is not necessary. Please remove.

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/thermal.py [hunk 20]
```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
```
Comment (jleveque) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/thermal.py:1]: Script is not executable; shebang is not necessary. Please remove.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/sonic-platform-modules-accton/debian/rules [hunk 21]
```diff
@@ -14,6 +14,7 @@ include /usr/share/dpkg/pkg-info.mk
 export INSTALL_MOD_DIR:=extra
 
 PYTHON         ?= python2
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-accton/debian/rules:16]: I see you're no longer building a Python 2 sonic-platform package. That's great. However, it appears you're still using Python 2 for something. Be aware that we will be removing Python 2 support from SONiC altogether in the near future.

Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/debian/rules:16]: Ok. Due to this PR is for API 2.0, so I remove python 2. For others use python2.7, we will check and test for each platform.  After done, we will send another PR for each Accton platform.

**Review by jleveque — CHANGES_REQUESTED** [review]
Please fix new conflict

**Comment by jostar-yang** [comment]
Done to fix new conflict.

**Review by jleveque — DISMISSED** [review]

**Comment by jostar-yang** [comment]
The PR is only for 202012br. So I need to close and resend new PR for 202012br.

**Pull request closed without merging**